src/clients/db/PatientRepository.ts

```diff
@@ -28,11 +34,72 @@ class PatientRepository extends Repository<Patient> {
     })
   }
 
+  async searchPaged(
+    text: string,
+    pageRequest: PageRequest = UnpagedRequest,
+    sortRequest: SortRequest = Unsorted,
+  ): Promise<Page<Patient>> {
+    const selector: any = {
+      $or: [
+        {
+          fullName: {
+            $regex: RegExp(text, 'i'),
+          },
+        },
+        {
+          code: text,
+        },
+      ],
+    }
+    sortRequest.sorts.forEach((s) => {
+      selector[s.field] = { $gt: null }
+    })
+
+    const result = await super
+      .search({
+        selector,
+        limit: pageRequest.size ? pageRequest.size + 1 : undefined,
+        skip:
+          pageRequest.number && pageRequest.size ? (pageRequest.number - 1) * pageRequest.size : 0,
+        sort:
+          sortRequest.sorts.length > 0
+            ? sortRequest.sorts.map((s) => ({ [s.field]: s.direction }))
+            : undefined,
+      })
+      .catch((err) => {
+        console.log(err)
+        return err
+      })
+
+    const pagedResult: Page<Patient> = {
+      content: result.slice(
+        0,
+        pageRequest.size
+          ? result.length < pageRequest.size
+            ? result.length
+            : pageRequest.size
+          : result.length,
+      ),
```

Comment:
I think that this should be handled by the database. Not sure that we should need to do a slice after we have a search returned.

Reply:
In the search query, one extra record is fetched.

e.g. if pagesize is 25, search query will search for 26 records. This is to know information about the next page.
But, while returning the result, only 25 records should be returned.
This line simply skips last record.